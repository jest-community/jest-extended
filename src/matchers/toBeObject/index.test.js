import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeObject', () => {
  it('passes when given an object', () => {
    expect({}).toBeObject();
  });

  each([[false], [''], [0], [() => {}], [undefined], [NaN]]).it('fails when not given an object: %s', given => {
    expect(() => expect(given).toBeObject()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeObject', () => {
  each([[false], [''], [0], [() => {}], [undefined], [NaN]]).it('passes when not given an object: %s', given => {
    expect(given).not.toBeObject();
  });

  it('fails when given an object', () => {
    expect(() => expect({}).not.toBeObject()).toThrowErrorMatchingSnapshot();
  });
});
