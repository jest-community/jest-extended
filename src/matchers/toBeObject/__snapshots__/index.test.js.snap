// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeObject fails when given an object 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeObject(</><dim>)</>

Expected value to not be an object, received:
  <red>{}</>"
`;

exports[`.toBeObject fails when not given an object:  1`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>\\"\\"</>"
`;

exports[`.toBeObject fails when not given an object: 0 1`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>0</>"
`;

exports[`.toBeObject fails when not given an object: NaN 1`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>NaN</>"
`;

exports[`.toBeObject fails when not given an object: false 1`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>false</>"
`;

exports[`.toBeObject fails when not given an object: undefined 1`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>[Function anonymous]</>"
`;

exports[`.toBeObject fails when not given an object: undefined 2`] = `
"<dim>expect(</><red>received</><dim>).toBeObject(</><dim>)</>

Expected value to be an object, received:
  <red>undefined</>"
`;
